### PR TITLE
Expose PR head ref name to prow jobs

### PR DESCRIPTION
**Pull request by dobesv**
Some tools really want to know the name of the branch being built (chromatic.com for example).

This could also be handy in some places where currently only the PR # is displayed; it would be nice to show the branch name as well as that is generally more user friendly.

This PR a initially a proposal - some more cleanup and testing is likely needed.  I haven't figured out how to build and run these tools myself as a test yet.

### CODE REVIEW

File: prow/pod-utils/downwardapi/jobspec.go [hunk 1]
```diff
@@ -98,13 +98,14 @@ const (
...
-	pullNumberEnv  = "PULL_NUMBER"
-	pullPullShaEnv = "PULL_PULL_SHA"
+	repoOwnerEnv       = "REPO_OWNER"
+	repoNameEnv        = "REPO_NAME"
+	pullBaseRefEnv     = "PULL_BASE_REF"
+	pullBaseShaEnv     = "PULL_BASE_SHA"
+	pullRefsEnv        = "PULL_REFS"
+	pullNumberEnv      = "PULL_NUMBER"
+	pullPullShaEnv     = "PULL_PULL_SHA"
+	pullHeadRefNameEnv = "PULL_HEAD_REF_NAME"
```
Comment (cjwagner) [inline prow/pod-utils/downwardapi/jobspec.go:108]: How about `PULL_HEAD_REF` for naming consistency?

**Review by cjwagner — COMMENTED** [review]
Generally looks good, but I think we should be a bit more consistent about naming.

Please also update the env documentation once this is in: https://docs.prow.k8s.io/docs/jobs/#job-environment-variables

File: prow/pod-utils/downwardapi/jobspec.go (hunk 1 above)
Comment (dobesv) [reply prow/pod-utils/downwardapi/jobspec.go]: Sounds fine to me, I updated it.

**Comment by dobesv** [comment]
> Generally looks good, but I think we should be a bit more consistent about naming.

No problem, I updated it as recommended.

> Please also update the env documentation once this is in

Yeah, I can do that once this is in.

I also updated the tests now, so they should pass if you run them.

Let me know if there's anything else I can do!

**Review by cjwagner — APPROVED** [review]

**Pull request merged**